internal/nginx/config/servers.go

```diff
@@ -8,29 +8,28 @@ import (
 	"sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/nginx/config/http"
-	"github.com/nginxinc/nginx-kubernetes-gateway/internal/nginx/config/template"
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state"
 )
 
 func executeServers(conf state.Configuration) []byte {
```

Comment:
After the refactoring, I'm not sure if we need some of the new types and functions.

For example, what if instead of in  `executeServers`:
```go
	t := newTemplate([]http.Server{})
	servers := createServers(conf.HTTPServers, conf.SSLServers)

	return t.execute(servers)
```

we had:
```go
        servers := createServers(conf.HTTPServers, conf.SSLServers)
	return execute(splitClientsTemplate, servers) // helper function to execute any template
```
and we also had the var with `splitClientsTemplate` either in this file or `servers_template` file (so that it is closed to where it is used).

Note that since we know we're generating the servers, we know which template to use -- so that's why I don't see the need for the template type and the type switch in `newTemplate`.